promql/engine_test.go

```diff
@@ -755,9 +755,8 @@ func TestRecoverEvaluatorRuntime(t *testing.T) {
 	//nolint:govet
 	a[123] = 1
 
-	if err.Error() != "unexpected error" {
-		t.Fatalf("wrong error message: %q, expected %q", err, "unexpected error")
-	}
+	unexpectedErr := errors.New("unexpected error")
+	testutil.ErrorsEqual(t, err, unexpectedErr)
```

Comment:
```suggestion
	testutil.ErrorsEqual(t, unexpectedErr, err)
```

Reply:
Hmm, I thought that the other way would be more usual - we usually write `if err != nil` and not `if nil != err`. Also all `testutil.Assert` conditions are written this way. 

While go allows writing yoda condition (`if nil != err`) we don't do this in `testutil.Assert`. Take, for example, this piece from tsdb/db_test.go:

```go
	testutil.Equals(t, 0, len(lns))
	testutil.Equals(t, 0, len(lvs))
	testutil.Assert(t, res.Next() == false, "")
```
In case of `Equals` the tested variable is on the right, but in `Assert` it's on the left. For me it just seems incosistent.

I don't mind changing it as you're suggesting but I would rather do it the other way - change all invocations of `testutil.Equals` to read as ordinary go condition.

Anyway, this may be a subject of another PR, so I'm doing the suggested change here.